integration-tests/src/test/java/oracle/weblogic/kubernetes/ItWDTModelNoServer.java

```diff
@@ -0,0 +1,466 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static oracle.weblogic.kubernetes.TestConstants.MANAGED_SERVER_NAME_BASE;
+import static oracle.weblogic.kubernetes.TestConstants.MII_BASIC_APP_NAME;
+import static oracle.weblogic.kubernetes.TestConstants.OPERATOR_RELEASE_NAME;
+import static oracle.weblogic.kubernetes.actions.TestActions.deleteClusterCustomResource;
+import static oracle.weblogic.kubernetes.actions.TestActions.getOperatorPodName;
+import static oracle.weblogic.kubernetes.assertions.TestAssertions.appAccessibleInPod;
+import static oracle.weblogic.kubernetes.utils.CommonMiiTestUtils.createMiiDomain;
+import static oracle.weblogic.kubernetes.utils.CommonMiiTestUtils.createMiiDomainAndVerify;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkPodReadyAndServiceExists;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkServiceDoesNotExist;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.testUntil;
+import static oracle.weblogic.kubernetes.utils.DomainUtils.createDomainInImageUsingWdt;
+import static oracle.weblogic.kubernetes.utils.DomainUtils.createDomainOnPvUsingWdt;
+import static oracle.weblogic.kubernetes.utils.DomainUtils.deleteDomainResource;
+import static oracle.weblogic.kubernetes.utils.DomainUtils.shutdownDomainAndVerify;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.createMiiImageAndVerify;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.createTestRepoSecret;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.imageRepoLoginAndPushImageToRegistry;
+import static oracle.weblogic.kubernetes.utils.LoggingUtil.checkPodLogContainsString;
+import static oracle.weblogic.kubernetes.utils.OperatorUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.PodUtils.checkPodDoesNotExist;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Test creating MII domain with different AdminServerName and Server settings in WDT model file.
+ * The test contains the following usecases:
+ * 1) AdminServerName is not specified and no Server section in the model file.
+ * 2) AdminServerName is not specified and no 'AdminServer' specified in Server section.
+ *    There is a server 'myadmin' specified in the Server instead.
+ * 3) AdminServerName is not specified and 'AdminServer' is specified in the Server section.
+ * 4) AdminServerName is not specified and 'adminserver' (all lower case) is specified in the Server section.
+ * 5) AdminServerName is specified and the named server is not set in the Server section.
+ * 6) Configured cluster defined but no managed server associated with it in the Server section.
+ * 7) Empty topology (no AdminServerName, Server and Cluster section)
+ */
```

Comment:
Is it possible to add a managed server named "payroll-admin" under Server section and not define it in cluster? I guess there is some logic which determines which server is admin and accordingly extract its info from model file.  I am just wondering if managed server has a keyword admin in its name, how does the logic behave!

Reply:
any server not belonging to a cluster will be standalone server,  There is no magic logic to determine which server is administration server,   the AdminServerName is the attribute to specify which server is the administration server.  If it is not specify, the it will have a server called 'AdminServer' coming from the WLS product's domain tamplate and automatically set the 'AdminServerName' as 'AdminServer'